columnar/src/backend/columnar/columnar_writer.c

```diff
@@ -283,6 +282,15 @@ ColumnarWriteRow(ColumnarWriteState *writeState, Datum *columnValues, bool *colu
 
 			chunkData->existsArray[columnIndex][chunkRowIndex] = true;
 
+			uint32 datumLength = att_addlength_datum(0, columnTypeLength, columnValues[columnIndex]);
+			uint32 datumLengthAligned = att_align_nominal(datumLength, columnTypeAlign);
+
+			if (datumLengthAligned >= (256UL << 20))
```

Comment:
we already have customers that are over the 256mb limit, how do we deal with that?

Reply:
not sure if we do or not, but to what degree would their tables already broken?

Reply:
It is limit per row/column. Just from previous PR github#221 it looks that they there could be some issues in database. For what is worth, this limit is not related to columnar storage. Same issue can be observed in heap itself

postgres=# create table t_heap(a text, b text);
CREATE TABLE
postgres=# insert into t_heap SELECT repeat('a', 1000000000), repeat('a', 1000000000);
ERROR:  invalid memory alloc request size 2000000056
postgres=# 

So, this is only related when we insert `columnar.chunk`. Complete row in that table should be less than 1GB in size.